hypervisor/qemu/qemu_cmd.go

```diff
@@ -0,0 +1,82 @@
+package qemu
+
+import (
+	"fmt"
+	"strconv"
+)
+
+type cmdLine struct {
+	args []string
+}
+
+func (c *cmdLine) appendArgs(args ...string) {
+	for _, arg := range args {
+		c.args = append(c.args, arg)
+	}
+}
+
+func (cmd *cmdLine) QemuBootCmd(m *Machine) []string {
+	cmd.appendArgs("-smp", strconv.Itoa(m.Cores), "-m", strconv.FormatUint(m.Memory, 10))
+	if m.Kvm {
+		cmd.appendArgs("-enable-kvm")
+	}
+	if len(m.Serial) > 0 {
+		cmd.appendArgs("-serial", fmt.Sprintf("unix:%s,server,nowait", m.Serial))
+	}
+	if len(m.Monitor) > 0 {
+		cmd.appendArgs("-monitor", fmt.Sprintf("unix:%s,server,nowait", m.Monitor))
+	}
+	if len(m.Vnc) > 0 {
+		cmd.appendArgs("-vnc", m.Vnc)
+	}
+	if len(m.Pidfile) > 0 {
+		cmd.appendArgs("-pidfile", m.Pidfile)
+	}
+	for _, d := range m.Drives {
+		drive := fmt.Sprintf("file=%s,format=%s", d.Image.Path, d.Image.Format)
+		if len(d.If) > 0 {
+			drive = fmt.Sprintf("%s,if=%s", drive, d.If)
+		}
+		cmd.appendArgs("-drive", drive)
+	}
```

Comment:
How do you manage the drive device order? It seems not to be deterministic if the OS drive has earlier drive number.